source/Octopus.Tentacle.Client/Scripts/ScriptExecutionOrchestrator.cs

```diff
@@ -282,17 +327,31 @@ await Task.Delay(scriptObserverBackOffStrategy.GetBackoff(++iteration), scriptEx
             var cancellationToken = CancellationToken.None;
             if (scriptServiceVersionToUse == ScriptServiceVersion.Version2)
             {
-                return await rpcCallExecutor.ExecuteWithRetries(
-                    RpcCall.Create<IClientScriptServiceV2>(nameof(IClientScriptServiceV2.CancelScript)),
-                    ct =>
-                    {
-                        return scriptServiceV2.CancelScript(new CancelScriptCommandV2(lastStatusResponse.Ticket, lastStatusResponse.NextLogSequence), new HalibutProxyRequestOptions(ct));
-                    },
-                    logger,
-                    // We don't want to abandon this operation as it is responsible for stopping the script executing on the Tentacle
-                    abandonActionOnCancellation: false,
-                    clientOperationMetricsBuilder,
-                    cancellationToken).ConfigureAwait(false);
+                ScriptStatusResponseV2 CancelScriptAction(CancellationToken ct)
+                {
+                    return scriptServiceV2.CancelScript(new CancelScriptCommandV2(lastStatusResponse.Ticket, lastStatusResponse.NextLogSequence), new HalibutProxyRequestOptions(ct));
+                }
+
+                if (rpcRetrySettings.RetriesEnabled)
+                {
+                    return await rpcCallExecutor.ExecuteWithRetries(
+                        RpcCall.Create<IClientScriptServiceV2>(nameof(IClientScriptServiceV2.CancelScript)),
+                        CancelScriptAction,
+                        logger,
+                        // We don't want to abandon this operation as it is responsible for stopping the script executing on the Tentacle
+                        abandonActionOnCancellation: false,
+                        clientOperationMetricsBuilder,
+                        cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    return rpcCallExecutor.Execute(
+                        RpcCall.Create<IClientScriptServiceV2>(nameof(IClientScriptServiceV2.CancelScript)),
+                        CancelScriptAction,
+                        abandonActionOnCancellation: true,
```

Comment:
`ExecuteWithRetries` uses `abandonActionOnCancellation: false,`. Did you mean to change this behavour to `abandonActionOnCancellation: true,` for `Execute`?